src/main/scala/model/Statistics.scala

```diff
@@ -0,0 +1,37 @@
+package model
+
+import model.entities.{BasicZombie, Entity, Zombie}
+
+object Statistics:
+
+  trait GameStats:
+    def entities: Seq[Entity]
+    def rounds: Int
+
+  trait ZombieStatsOps:
+    stats: GameStats =>
+    def getZombies: Seq[Zombie] =
+      stats.entities.foldLeft(List.empty)((acc, e) => e match {
+        case zombie: Zombie => acc :+ zombie
+        case _ => acc
+      })
+    def playWave(seq: Seq[Zombie]): GameStatistics =

```

Comment:
A more relevant statistic would be the number of killed zombies. Also maybe the method name could be improved

Reply:
I agree with you: playWave makes no sense and just introduces complications into the code.